overlord/fdestate/fdemgr.go

```diff
@@ -184,6 +193,33 @@ func (m *FDEManager) isFunctional() error {
 	return m.initErr
 }
 
+var ensureUniqueContainerRoles = ensureUniqueContainerRolesImpl
+
+func ensureUniqueContainerRolesImpl(m *FDEManager) error {
+	if err := m.isFunctional(); err != nil {
+		// skip
+		return nil
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	containers, err := m.GetEncryptedContainers()
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(containers); i++ {
+		for j := i + 1; j < len(containers); j++ {
+			if containers[i].ContainerRole() == containers[j].ContainerRole() {
+				m.state.Warnf("container roles should map to one volume only: container role %q maps to %s and %s", containers[i].ContainerRole(), containers[i].DevPath(), containers[j].DevPath())
+			}
+		}
+	}
```

Comment:
This looks like a post-condition of GetEncryptedContainer. Maybe it should always run there. But the logging would not work.
This is fine, but I want to just add a comment now. And I think about it.

Reply:
This is a warning not a logger, warnings are not repeated for the same message so actually I think it can simply go there and simplify the ensure loop into a no-op again

Reply:
thanks, this is now a lot simpler as a post-condition. Currently it cannot be tested but acts as a reminder with a comment explaining this.